CHANGELOG.md

```diff
@@ -21,6 +21,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 ### Changed
 
 - [#5255](https://github.com/thanos-io/thanos/pull/5296) Query: Use k-way merging for the proxying logic. The proxying sub-system now uses much less resources (~25-80% less CPU usage, ~30-50% less RAM usage according to our benchmarks). Reduces query duration by a few percent on queries with lots of series.
+- [#5641](https://github.com/thanos-io/thanos/pull/5641) Store: Remove hardcoded labels in shard matcher. Query: Inject unshardable le label in query analyzer.
```

Comment:
Nit: Is the query change relevant to end users? If so we can keep it, but I'd just add it as a separate point, read a bit better.

Reply:
Done. I split it into two separate points.